client/components/root-child/index.jsx

```diff
@@ -46,7 +47,13 @@ export default class RootChild extends React.Component {
 		// Context is lost when creating a new render hierarchy, so ensure that
 		// we preserve the context that we care about
 		if ( this.context.store ) {
-			content = <ReduxProvider store={ this.context.store }>{ content }</ReduxProvider>;
+			content = (
+				<ReduxProvider store={ this.context.store }>
+					<CalypsoWPDataProvider calypsoStore={ this.context.store }>
+						{ content }
+					</CalypsoWPDataProvider>
```

Comment:
I think this needs to go to `client/layout` instead, at least for the purpose of this PR. `RootChild` is mostly used by modals, whereas the color scheme picker (like most components) is a direct descendant of `client/layout`. I _think_ that in this PR, `withSelect()` and `withDispatch()` are currently picking up a 'global' registry instance (inside `@wp/data`).

I _think_ I got this to work so it picks up the correct registry in my PR, #26930.

Reply:
Thanks for the tip. I'll adjust this one too.

Reply:
Would it make sense to have the WP Data Provider in both places? Both here and in ReduxWrappedLayout?

Reply:
Addressed in e9be45c

Reply:
> Would it make sense to have the WP Data Provider in both places? Both here and in ReduxWrappedLayout?

Eventually, we'll need that Provider at the top of all component hierarchies contain components wrapped in `withSelect` and/or `withDispatch`, but for the sake of this PR, wrapping only in `ReduxWrappedLayout` should be enough.

(We might end up writing a HOC that contains both the Redux and Calypso WP Data Provider, and wrap all component hierarchies in that.)